core/src/main/java/org/infinispan/interceptors/distribution/L1LastChanceInterceptor.java

```diff
@@ -76,40 +76,50 @@ public void start() {
    }
 
    public CompletableFuture<Void> visitDataWriteCommand(InvocationContext ctx, DataWriteCommand command, boolean assumeOriginKeptEntryInL1) throws Throwable {
-      Object returnValue = ctx.forkInvocationSync(command);
-      Object key;
-      if (shouldUpdateOnWriteCommand(command) && command.isSuccessful() &&
-            cdl.localNodeIsOwner((key = command.getKey()))) {
-         if (trace) {
-            log.trace("Sending additional invalidation for requestors if necessary.");
+      ctx.onReturn((ctx1, command1, rv, throwable) -> {
+         if (throwable != null)
+            throw throwable;
+
+         Object key;
+         if (shouldUpdateOnWriteCommand(command) && command.isSuccessful() &&
```

Comment:
is there a case where the ```command != command1``` or ```ctx != ctx1```? if not, I would remove the arguments from the return handler. otherwise, you need to update the code use the ```command1``` :)